dotnet/src/SemanticKernel.Abstractions/IKernel.cs

```diff
@@ -43,43 +42,21 @@ public interface IKernel
     IReadOnlySkillCollection Skills { get; }
 
     /// <summary>
-    /// Build and register a function in the internal skill collection, in a global generic skill.
-    /// </summary>
-    /// <param name="functionName">Name of the semantic function. The name can contain only alphanumeric chars + underscore.</param>
-    /// <param name="functionConfig">Function configuration, e.g. I/O params, AI settings, localization details, etc.</param>
-    /// <returns>A C# function wrapping AI logic, usually defined with natural language</returns>
-    ISKFunction RegisterSemanticFunction(
-        string functionName,
-        SemanticFunctionConfig functionConfig);
-
-    /// <summary>
-    /// Build and register a function in the internal skill collection.
+    /// Import a set of functions from the given skill. The functions must have the `SKFunction` attribute.
+    /// Once these functions are imported, the prompt templates can use functions to import content at runtime.
     /// </summary>
-    /// <param name="skillName">Name of the skill containing the function. The name can contain only alphanumeric chars + underscore.</param>
-    /// <param name="functionName">Name of the semantic function. The name can contain only alphanumeric chars + underscore.</param>
-    /// <param name="functionConfig">Function configuration, e.g. I/O params, AI settings, localization details, etc.</param>
-    /// <returns>A C# function wrapping AI logic, usually defined with natural language</returns>
-    ISKFunction RegisterSemanticFunction(
-        string skillName,
-        string functionName,
-        SemanticFunctionConfig functionConfig);
+    /// <param name="skillInstance">Instance of a class containing functions</param>
+    /// <param name="skillName">Name of the skill for skill collection and prompt templates. If the value is empty functions are registered in the global namespace.</param>
```

Comment:
"If the value is empty"... why not if it's null?  Wondering why we have `string skillName = ""` instead of `string? skillName = null`.

Reply:
I don't think there's a strong reason. We've been back and forth with nullability checks about twenty times (not kidding) because we keep switching language version/.net version. It's kind of annoying in fact

Reply:
We've had a mix of empty and null default string args throughout the codebase. I've been giving guidance in PRs to use "string? arg = null", as this is a stronger indication that the string was not provided at all, rather than a string was provided that happens to be empty (and, in some cases, is valid).